### PR TITLE
KNIME-1429: Added preferences to control RDKit 2D Depiction

### PR DESCRIPTION
**Pull request by manuelschwarze**
**Added preferences to control RDKit 2D Depiction**

Rewrite of RDKit value renderer based on improvements in RDKit binaries by Paolo Tosco.
Added configuration capabilities based on JSON settings.
Provided default settings file.

Note: This branch contains changes tracked in #115. It will be merged together into master.

### CODE REVIEW

**Comment by manuelschwarze** [comment]
@greglandrum, could you please add Paolo Tosco as reviewer. I cannot pick him from the list, I guess you as admin have to configure this somehow... Thanks!

**Comment by greglandrum** [comment]
@ptosco could you please take a look at this one?

**Comment by greglandrum** [comment]
> @greglandrum, could you please add Paolo Tosco as reviewer. I cannot pick him from the list, I guess you as admin have to configure this somehow... Thanks!

You can just @ mention him (which I just did) and he'll get a notification

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java [hunk 1]
```diff
@@ -68,32 +66,34 @@
...
  * @author Thorsten Meinl, University of Konstanz
  * @author Manuel Schwarze, Novartis
+ * @author Paolo Tosco, Novartis
  */
 public class RDKitMolValueRenderer extends AbstractPainterDataValueRenderer
 implements SvgProvider {
 
 	/**
-	 * Factory for {@link RDKitMolValueRenderer}.
+	 * Factory for {@link RDKitMolValueRenderer2}.
```
Comment (ptosco) [inline org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java]: Is this change intentional?

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java [hunk 2]
```diff
@@ -382,12 +379,72 @@ else if (m_svgDocument != null) {
...
+					RDKFuncs.updateDrawerParamsFromJSON(molDrawing, strJsonConfig);
+				}
+				
+				MolDrawOptions opts = molDrawing.drawOptions();
+				
+				// We've already prepared the molecule appropriately, so don't try again
+				opts.setPrepareMolsBeforeDrawing(false);
+				
+				// Apply old config, if no JSON config is provided (before RDKit Types version 4.6.0)
+				if (strJsonConfig == null) {
```
Comment (ptosco) [inline org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java:409]: Please note that the JSON file that I sent you yesterday does not do anything about stereo annotation.

**Review by ptosco — APPROVED** [review]
LGTM, I only put in a couple of comments.

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java (hunk 1 above)
Comment (manuelschwarze) [reply org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java]: Nope, thanks for spotting it, it is a left over from some try. Will fix this.

**Review by greglandrum — APPROVED** [review]
LGTM.
I ran the tests with KNIME 4.5 on my linux box and things look good.

**Comment by greglandrum** [comment]
Well, I shouldn't say that they "look" good since I don't particularly care for @ptosco's favorite drawing settings (the lines are too thin... bonds are *important*), but that's a matter of taste and not worth arguing about. :-)

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java (hunk 2 above)
Comment (manuelschwarze) [reply org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java:409]: Thanks for pointing this out. I think we should set stereo annotation to true then in our defaults to be as close as possible to the old drawings from its content. They are supposed to just look nicer :-) 
@ptosco, can you tell me the exact setting I should put in to the JSON, like programmatically "opts.setAddStereoAnnotation(true);" ? Is there any other setting that we should add to be close to the old default settings?

**Comment by ptosco** [comment]
@greglandrum The drawing settings are optimized for web apps that display lots of tiny molecules, and there it is important to keep bonds thin and sharp or all aromatic rings look like black blobs. If you think they look too thin in KNIME it is perfectly fine to make them thicker, but the default 2.0 width is probably too much also for KNIME.

**Comment by ptosco** [comment]
@greglandrum what's your take on stereo labels? My opinion is that they also clutter the picture a lot. Any reason why they are switched on in KNIME whereas in RDKit they are off by default?
FWIW, also Avalon did not show stereo labels; this is quinine drawn by Avalon:
image

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java (hunk 1 above)
Comment (manuelschwarze) [reply org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java]: Done

File: org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java (hunk 2 above)
Comment (manuelschwarze) [reply org.rdkit.knime.types/rdkit-chemsrc/org/rdkit/knime/types/RDKitMolValueRenderer.java:409]: Done, added stereo annotation also in new default settings, increased a bit the bond line width to make Greg happy :-).

**Comment by greglandrum** [comment]
> @greglandrum what's your take on stereo labels? My opinion is that they also clutter the picture a lot. Any reason why they are switched on in KNIME whereas in RDKit they are off by default?

I've heard from numerous people that they like having the stereo labels. I agree that they can clutter things, but it seems worth it to many people. But I don't have a particularly strong opinion.
Enhanced stereo labels, on the other hand, should definitely be displayed by default (even though they aren't in the RDKit at the moment). Those provide additional information which you can't determine just by looking at the structure itself.
